File: pizzastore/src/main/java/com/mesoneer/pizzastore/entity/Order.java
package com.mesoneer.pizzastore.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "orders")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    private LocalDateTime bookingDate;
    private String customerName;
    private String phoneNumber;
    private String deliveryTo;

    @Enumerated(EnumType.ORDINAL)
    private Status status;
    private float finalAmount;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    private List<OrderItem> orderItems;

}

File: pizzastore-frontend1/src/app/model/order.ts
import { OrderItem } from "./orderItem";

export interface Order {
  // id: number;
  customerName: string;
  phoneNumber: string;
  deliveryTo: string;
  finalAmount: number;
  orderItems: OrderItem[];
  status: string;

  // constructor(customerName: string, phoneNumber: string, deliveryTo: string, finalAmount: number, orderItems: OrderItem[]) {
  //   this.customerName = customerName;
  //   this.phoneNumber = phoneNumber;
  //   this.deliveryTo = deliveryTo;
  //   this.finalAmount = finalAmount;
  //   this.orderItems = orderItems;
  // }
}